test_socket.py

```diff
@@ -0,0 +1,140 @@
+import pytest
+import socket
+import string
+import email.utils
+
+
+@pytest.fixture(scope='session')
+def connection():
+    ADDR = ('127.0.0.1', 9992)
+    client = socket.socket(
+        socket.AF_INET, socket.SOCK_STREAM, socket.IPPROTO_IP
+    )
+    client.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+    client.connect(ADDR)
+    return client
+
+
+def test_wrongbreak():
+    ADDR = ('127.0.0.1', 9992)
+    client = socket.socket(
+        socket.AF_INET, socket.SOCK_STREAM, socket.IPPROTO_IP
+    )
+    client.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+    client.connect(ADDR)
```

Comment:
lines 19-24 are a direct repeat of the stuff that is in your fixture above.  Why repeat it?  just use the fixture.